drivers/net/oa_tc6/oa_tc6.c

```diff
@@ -0,0 +1,2176 @@
+ *   Obtain parity of a 32-bit word.
+ *
+ * Input Parameters:
+ *   word - 32-bit word, subject to the parity calculation
+ *
+ * Returned Value:
+ *   If the parity of the word is even, zero is returned.
+ *   Otherwise one is returned.
+ *
+ ****************************************************************************/
+
+static int oa_tc6_get_parity(uint32_t word)
+{
+  /* www-graphics.stanford.edu/~seander/bithacks.html */
+
+  word ^= word >> 1;
+  word ^= word >> 2;
+  word = (word & 0x11111111u) * 0x11111111u;
+  return (word >> 28) & 1;
+}
+
+/****************************************************************************
+ * Name: oa_tc6_poll_footer
+ *
+ * Description:
+ *   Poll a data transaction chunk footer.
+ *
+ * Input Parameters:
+ *   priv   - pointer to the driver-specific state structure
+ *   footer - pointer to a 32-bit footer destination variable
+ *
+ * Returned Value:
+ *   On a successful transaction OK is returned, otherwise ERROR is returned.
+ *
+ ****************************************************************************/
+
+static int oa_tc6_poll_footer(FAR struct oa_tc6_driver_s *priv,
+                              FAR uint32_t *footer)
+{
+  uint8_t *txbuf = priv->txbuf;
+  uint8_t *rxbuf = priv->rxbuf;
+  uint32_t header;
+
+  header =   (1 << OA_TC6_DNC_POS)   /* Data Not Control */
+           | (1 << OA_TC6_NORX_POS); /* No Read */
+
+  if (oa_tc6_exchange_chunk(priv, txbuf, rxbuf, header, footer))
+    {
+      return ERROR;
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: oa_tc6_exchange_chunk
+ *
+ * Description:
+ *   Send a data chunk to MAC-PHY and simultaneously receive chunk.
+ *
+ *   Computing header parity, checking footer parity, converting to proper
+ *   endianness and setting DNC flag is done by this function.
+ *
+ * Input Parameters:
+ *   priv   - pointer to the driver-specific state structure
+ *   txbuf  - buffer with transmit chunk data
+ *   rxbuf  - buffer to save the received chunk to
+ *   header - header controlling the transaction
+ *   footer - pointer to a 32-bit value for the footer
+ *
+ * Returned Value:
+ *   On a successful transaction OK is returned, otherwise ERROR is returned.
+ *
+ ****************************************************************************/
+
+static int oa_tc6_exchange_chunk(FAR struct oa_tc6_driver_s *priv,
+                                 FAR uint8_t *txbuf, FAR uint8_t *rxbuf,
+                                 uint32_t header, FAR uint32_t *footer)
+{
+  header |= (1 << OA_TC6_DNC_POS);
+  header |= (!oa_tc6_get_parity(header) << OA_TC6_P_POS);
+  header = htobe32(header);
+
+  ((uint32_t *)txbuf)[0] = header;
+
+  oa_tc6_select_spi(priv);
+  SPI_EXCHANGE(priv->spi, txbuf, rxbuf, OA_TC6_CHUNK_SIZE(priv));
+  oa_tc6_deselect_spi(priv);
+
+  *footer = *((uint32_t *)(&rxbuf[priv->config->chunk_payload_size]));
+
+  *footer = be32toh(*footer);
+  if (!oa_tc6_get_parity(*footer) ||
```

Comment:
I would consider storing the result in a temporary variable instead of calling these functions twice.